kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -131,18 +132,27 @@ instance SimplifyRuleLHS (ReachabilityRule Variable) where
         (fmap . fmap) AllPath $ simplifyRuleLhs rule
 
 simplifyClaimRule
-    :: (MonadSimplify simplifier, InternalVariable variable)
+    :: forall simplifier variable
+    .  MonadSimplify simplifier
+    => InternalVariable variable
     => RulePattern variable
     -> simplifier (MultiAnd (RulePattern variable))
-simplifyClaimRule rule@(RulePattern _ _ _ _ _) =
-    simplifyRuleLhs rule
-        { RulePattern.left =
-            mkAnd
-                left
-                -- Add the predicate to the left term so that it gets simplified
-                -- by the rule pattern simplifier.
-                (mkAnd (Predicate.unwrapPredicate requires) (mkCeil_ left))
-        , RulePattern.requires = makeTruePredicate_
-        }
+simplifyClaimRule =
+    fmap MultiAnd.make
+    . Branch.gather
+    . Lens.traverseOf RulePattern.leftPattern worker
   where
-    RulePattern {left, requires} = rule
+    worker, simplify, simplifyWithSolver
+        :: Pattern variable
+        -> BranchT simplifier (Pattern variable)
+    worker =
+        (return . Pattern.requireDefined)
+        >=> simplify
+        >=> simplifyWithSolver
```

Comment:
We usually call `SMT.Evaluator.filterMultiOr` on the result of `simplifyTopConfiguration` instead of, say, `simplifyConditionsWithSmt` because we assume that the condition is already fully simplified (implied by the function name).